File: bungeecord/src/main/java/ru/privatenull/pnauth/bungee/BungeeMessages.java
package ru.privatenull.pnauth.bungee;

import net.md_5.bungee.api.chat.TextComponent;
import net.md_5.bungee.api.chat.BaseComponent;
import net.md_5.bungee.api.ChatColor;
import net.kyori.adventure.text.Component;
import ru.privatenull.pnauth.message.MessageFormat;
import ru.privatenull.pnauth.message.MessageRenderers;

final class BungeeMessages {
    private BungeeMessages() {
    }

    static BaseComponent component(String message, MessageFormat format) {
        return TextComponent.fromArray(components(message, format));
    }

    static BaseComponent[] components(String message, MessageFormat format) {
        if (format == MessageFormat.JSON) {
            try {
                return net.md_5.bungee.chat.ComponentSerializer.parse(message == null ? "" : message);
            } catch (RuntimeException ignored) {
                // Fall through to inert legacy text for a malformed administrator template.
            }
        }
        String legacy = MessageRenderers.toLegacy(message == null ? "" : message,
                format == null ? MessageFormat.LEGACY : format);
        return TextComponent.fromLegacyText(ChatColor.translateAlternateColorCodes('&', legacy));
    }

    static Component adventureComponent(String message, MessageFormat format) {
        String value = message == null ? "" : message;
        MessageFormat selected = format == null ? MessageFormat.LEGACY : format;
        String legacy = MessageRenderers.toLegacy(value, selected);
        String plain = ChatColor.stripColor(ChatColor.translateAlternateColorCodes('&', legacy));
        return Component.text(plain == null ? "" : plain);
    }
}

File: bungeecord/src/main/kotlin/ru/privatenull/pnauth/bungee/BungeeMessages.kt
@file:Suppress("DEPRECATION")
package ru.privatenull.pnauth.bungee

import net.kyori.adventure.text.Component
import net.md_5.bungee.api.chat.BaseComponent
import net.md_5.bungee.api.chat.TextComponent
import net.md_5.bungee.chat.ComponentSerializer
import ru.privatenull.pnauth.message.MessageFormat
import ru.privatenull.pnauth.message.MessageComponents

internal object BungeeMessages {

    @JvmStatic
    fun component(message: String?, format: MessageFormat?): BaseComponent {
        return TextComponent.fromArray(*components(message, format))
    }

    @JvmStatic
    fun components(message: String?, format: MessageFormat?): Array<BaseComponent> {
        val component = MessageComponents.deserialize(message, format)
        return ComponentSerializer.parse(MessageComponents.serializeJson(component))
    }

    @JvmStatic
    fun adventureComponent(message: String?, format: MessageFormat?): Component {
        return MessageComponents.deserialize(message, format)
    }
}
